src/ui_fsmenu/addons/manager.cc

```diff
@@ -849,9 +913,48 @@ bool AddOnsCtrl::handle_key(bool down, SDL_Keysym code) {
 	return UI::Window::handle_key(down, code);
 }
 
+void AddOnsCtrl::erase_remote(std::shared_ptr<AddOns::AddOnInfo> a) {
+	for (auto& pointer : remotes_) {
+		if (pointer == a) {
+			pointer = remotes_.back();
+			remotes_.pop_back();
+			return;
+		}
+	}
+	NEVER_HERE();
+}
+
 void AddOnsCtrl::refresh_remotes() {
+	UI::ProgressWindow progress(this, "", "");
+	const std::string step_message = _("Fetching add-ons (%.1f%%)");
+
 	try {
-		remotes_ = net().refresh_remotes();
+		progress.step(_("Connecting to the server…"));
+
+		std::vector<std::string> names = net().refresh_remotes(net().is_admin());
```

Comment:
Suggestion for the future: Allow showing all remotes to normal users, but show a warning message and mark possibly incompatible add-ons.